src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -179,23 +179,20 @@ public CSVFileScanFragment(
   @Override
   protected TupleBatch fetchNextReady() throws IOException, DbException {
     long lineNumberBegin = lineNumber;
-    boolean unfinishedEncapsulatedTokenRow = false;
+    boolean truncatedQuotedField = false;
 
     while ((buffer.numTuples() < TupleBatch.BATCH_SIZE) && !flagAsIncomplete) {
       lineNumber++;
       if (parser.isClosed()) {
         break;
       }
 
-      if (unfinishedEncapsulatedTokenRow) {
+      if (truncatedQuotedField) {
         onLastRow = true;
       }
 
-      try {
-        if (!onLastRow) {
-          record = iterator.next();
-        }
-      } catch (Exception e) {
+      if (!onLastRow) {
+        record = iterator.next();
```

Comment:
While reviewing the logic for handling unclosed quotes, I noticed that the meaning of `truncatedQuotedField` might be clearer if it were renamed to something like `nextRecordTruncatedWithinQuotedField` (that's pretty verbose, obviously--hope you can do better). The idea I want to convey is that it's not the *current* record that's truncated, but the *next* record.